src/components/Routes/Rockets.jsx

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { setRockets } from '../../redux/rocketsSlice';
+import RocketItem from '../RocketItem';
+
+function Rockets() {
+  const rockets = useSelector((state) => state.rockets);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (rockets.length === 0) {
+      fetch('https://api.spacexdata.com/v3/rockets')
+        .then((response) => response.json())
+        .then((data) => {
+          dispatch(setRockets(data));
+        });
+    }
+  }, [rockets, dispatch]);
+
+  return (
+    <div>
+      <br />
+      <ul>
+        {rockets.map((rocket) => (
+          <RocketItem key={rocket.id} rocket={rocket} />
+        ))}
+      </ul>
+    </div>
```

Comment:
- [ ] Good job creating the rocket components, I noticed an unusual result when it comes to this component. Whenever the Rockets page is refreshed, all the state is reset, kindly ensure that once the user refreshes the Rockets page, the state does not have to be reset. This applies to the mission page too.

Reply:
Hello @Olamarx , regarding this. In your previous comment says we shouldn't preserve user data, so when the page is reloaded the state will inevitably reset.
Furthermore as seen in this picture:
![image](https://github.com/tomasesquivelgc/Space-Travelers-Hub/assets/117373513/f7a89cd1-a39a-4c2a-8081-910e24142e79)
Even after several rotations and clicks the data is only fetched once.
It becomes clear on the [LIVE VERSION](https://guileless-halva-645039.netlify.app) that the page suffers no stutters or reloading.